WordPress/gutenberg
# Improve tools panel accessibility

## What?
A few fixes for https://github.com/WordPress/gutenberg/issues/40460

Addresses some accessibility issues for the `ToolsPanel` component:
- Use correct role for the Reset button—it's now a `menu-item` and doesn't have `aria-selected` specified.
- Don't close the dropdown when using the menu, this is to avoid unexpected focus shift.
- Improve button labelling. The previous button just had 'View options', which makes it hard to know whether the options are for color, dimensions without exploring the surrounding DOM.
- Use `speak` to announce the outcome of selecting a menu item in the options dropdown
- Show the 'reset all' button as disabled when there's nothing to reset

Also:
- Fix an issue for icon `Button`s, where the tooltip displayed the description instead of the label.

## Testing Instructions
(with a screenreader active)
1. Insert a paragraph block
2. Tab to the block inspector and continue tabbing until you get to 'Color options'
3. Expand the color options and select 'Show link'
4. You should hear an announcement 'Link now visible'. The menu should stay open.
5. Arrow down to the Reset All button and press it
6. You should hear an announcement 'All options reset'
7. Close the menu
8. Tab down to the font size radio buttons and select a font size
9. Shift tab to 'Typography options' and open the dropdown
11. Select 'Reset font size', an announcement 'Font size reset to default' should be heard
